### PR TITLE
feature_notify_inode_expire_only

### PR DESCRIPTION
**Pull request by HereThereBeDragons**
Related to Issue #707 

This PR extends the unoffical patch provided by Miklos Szeredi  in this [conversation](https://sourceforge.net/p/fuse/mailman/fuse-devel/thread/CAJfpegs8VKu45Jsv%2B-qAWwz2A%2BYpxCtxrFJr%2BA-b1g%2BzKsf%3DOA%40mail.gmail.com/#msg37703271) by 
- hopefully allowing the correct discovery if `fuse_lowlevel_notify_expire_entry` is available or not 
- adding some comments/description

### CODE REVIEW

**Review by Nikratio — CHANGES_REQUESTED** [review]
Thanks! Would it be possible to add a test case for this? (Similar to https://github.com/libfuse/libfuse/blob/master/test/test_ctests.py#L51)

**Comment by HereThereBeDragons** [comment]
Hi,
So I extended the tests as suggested. However, the current simple file system used for inval_entry will not allow us to correctly test if the functionality is available: right now the test would pass if you have this libfuse patch, but not the kernel patch - it would just work as the "notify_inval_entry" as the new flag is just ignored by the kernel.

Using notify_expire_entry makes only a visible difference for:
- mount on top
- depending on the implementation?: changing symlinks from short to long symlinks

so this needs a bit more sophisticated file system.

if you have any suggestions what would be the easiest to extend, please let me know 
(`passthrough_ll` might be an option but would need to be extended to track dentries)

ps. if you want to have a look: i pushed it to a different [branch](https://github.com/HereThereBeDragons/libfuse/tree/WIP_testing_notify_inode_expire_only) as it doesnt help us here

**Comment by Nikratio** [comment]
My stance is that a simple test is better than no test. Even if the test passes if the kernel test is missing, it will be able to catch some problems. So let's not make the perfect the enemy of the good.

File: test/test_examples.py [hunk 1]
```diff
@@ -329,13 +329,17 @@ def test_fn(name):
         umount(mount_process, mnt_file)
 
 
+names = ['notify_inval_entry',]
+if fuse_proto >= (7,38):
```
[the remark this answers is not in the document]
Comment (HereThereBeDragons) [reply test/test_examples.py:333]: (feel free to remove it - or let me know and i'll do it)

**Comment by HereThereBeDragons** [comment]
Hi @Nikratio,
Could you please let me briefly know if there is anything else I need to do for this PR in order to be accepted?
Not that I forgetting something from my side.

File: example/notify_expire_entry.c [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+  Copyright (C) 2016 Nikolaus Rath <Nikolaus@rath.org>
+
+  This program can be distributed under the terms of the GNU GPLv2.
+  See the file COPYING.
+*/
+
+/** @file
+ *
+ * This example implements a file system with a single file whose
+ * file name changes dynamically to reflect the current time.
```
Comment (Nikratio) [inline example/notify_expire_entry.c:12]: This looks very close to the existing notify_inval_entry.c example. Would it be possible to instead modify the existing example to add a --expire flag (to use the different notification function) to reduce code duplication?

File: test/test_examples.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Nikratio) [reply test/test_examples.py:333]: Well, if it's not needed then let's remove it.

**Review by Nikratio — CHANGES_REQUESTED** [review]
Sorry for the delay, I somehow missed that this has been updated.

Thank you for adding the test case. I just have two comments, but generally this looks good.

**Comment by Nikratio** [comment]
Thanks! One last request: could you please combine the last two commits (so that we don't create a new file just to remove it again), and get rid of the remaining whitespace-only changes?

Thank you for your patience, and apologies for the long review time!

**Comment by HereThereBeDragons** [comment]
Done. Let me know if there is anything else missing.
And np, everything is slow and quiet around Christmas and New Year.

**Pull request merged**